docs/source/en/basics/router.md

```diff
@@ -0,0 +1,328 @@
+  ctx.body = `body: ${JSON.stringify(ctx.request.body)}`;
+};
+
+// simulate a post request.
+// curl -X POST http://127.0.0.1:7001/form --data '{"name":"controller"}' --header 'Content-Type:application/json'
+```
+
+> P.S.:
+
+> If you perform a POST request directly, an **error** will occur: 'secret is missing'. This error message comes from [koa-csrf/index.js#L69](https://github.com/koajs/csrf/blob/2.5.0/index.js#L69).
+
+> **Reason**: the framework verifies the CSFR value specially for form POST requests, so please submit the CSRF key as well when you submit a form. For more detail refer to [Keep Away from CSRF Threat](https://eggjs.org/zh-cn/core/security.html#å®‰å…¨å¨èƒcsrfçš„é˜²èŒƒ)
+
+> **Note**: the verification is performed because the framework builds in a security plugin [egg-security](https://github.com/eggjs/egg-security) that provides some default security practices and this plugin is enabled by default. In case you want to disable some security protections, just configure the enable attribute of if to be false.
+
+> "Unless you know the consequence consciously, it's recommended not to disable functions provided by the security plugin"
+
+> Here we do the config temporarily in `config/config.default.js` for an example
+```
+exports.security = {
+  csrf: false
+};
+```
+
+### Form Verification
+
+```js
+// app/router.js
+module.exports = app => {
+  app.post('/user', 'user');
+};
+
+// app/controller/user.js
+const createRule = {
+  username: {
+    type: 'email',
+  },
+  password: {
+    type: 'password',
+    compare: 're-password',
+  },
+};
+
+exports.create = function* (ctx) {
+  // throws exceptions if the verification fails
+  ctx.validate(createRule);
+  ctx.body = ctx.request.body;
+};
+
+// curl -X POST http://127.0.0.1:7001/user --data 'username=abc@abc.com&password=111111&re-password=111111'
+```
+
+### Redirection
+
+#### Internal Redirection
+
+```js
+// app/router.js
+module.exports = app => {
+  app.get('index', '/home/index', 'home.index');
+  app.redirect('/', '/home/index', 302);
+};
+
+// app/controller/home.js
+exports.index = function* (ctx) {
+  ctx.body = 'hello controller';
+};
+
+// curl -L http://localhost:7001
+```
+
+#### External Redirection
+
+```js
+// app/router.js
+module.exports = app => {
+  app.get('/search', 'search');
+};
+
+// app/controller/search.js
+module.exports = function* () {
+  const type = this.query.type;
+  const q = this.query.q || 'nodejs';
+
+  if (type === 'bing') {
+    this.redirect(`http://cn.bing.com/search?q=${q}`);
+  } else {
+    this.redirect(`https://www.google.co.kr/search?q=${q}`);
+  }
+};
+
+// curl http://localhost:7001/search?type=bing&q=node.js
+// curl http://localhost:7001/search?q=node.js
+```
+
+### Using Middleware
+
+A middleware can be used to change the request parameter of one kind of user requests to uppercase.
+Here we show how to use the middleware briefly, referring to [Middleware](./middleware.md) for detail.
```

Comment:
suggestion:
'Here we just briefly explain how to use the middleware, refer to ...'